Rename boxing/unboxing files and utilities

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35488 Remove functor factories in KernelFunction
* **#35411 Rename boxing/unboxing files and utilities**

The file and class names in ATen/core/boxing were quite confusing.
Let's rename them for readability.

Also move function schema inference out of the boxing logic into op_registration.h where it belongs.

This PR is just moving code around and renaming things. No logic changes.

Differential Revision: [D20653621](https://our.internmc.facebook.com/intern/diff/D20653621/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D20653621/)!

smessmer committed:
Rename boxing/unboxing files and utilities

The file and class names in ATen/core/boxing were quite confusing.
Let's rename them for readability.

Also move function schema inference out of the boxing logic into op_registration.h where it belongs.

Differential Revision: [D20653621](https://our.internmc.facebook.com/intern/diff/D20653621/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D20653621/)!

[ghstack-poisoned]

smessmer committed:
Update on "Rename boxing/unboxing files and utilities"


The file and class names in ATen/core/boxing were quite confusing.
Let's rename them for readability.

Also move function schema inference out of the boxing logic into op_registration.h where it belongs.

This PR is just moving code around and renaming things. No logic changes.

Differential Revision: [D20653621](https://our.internmc.facebook.com/intern/diff/D20653621/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D20653621/)!

[ghstack-poisoned]

smessmer committed:
Update on "Rename boxing/unboxing files and utilities"


The file and class names in ATen/core/boxing were quite confusing.
Let's rename them for readability.

Also move function schema inference out of the boxing logic into op_registration.h where it belongs.

This PR is just moving code around and renaming things. No logic changes.

Differential Revision: [D20653621](https://our.internmc.facebook.com/intern/diff/D20653621/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D20653621/)!

[ghstack-poisoned]

ezyang commented:
Requesting to delay this refactor until https://github.com/pytorch/pytorch/pull/35061 lands

ezyang commented:
If we're renaming things, can we do the thing where file name exactly matches the class name in the file?

smessmer committed:
Update on "Rename boxing/unboxing files and utilities"


The file and class names in ATen/core/boxing were quite confusing.
Let's rename them for readability.

Also move function schema inference out of the boxing logic into op_registration.h where it belongs.

This PR is just moving code around and renaming things. No logic changes.

Differential Revision: [D20653621](https://our.internmc.facebook.com/intern/diff/D20653621/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D20653621/)!

[ghstack-poisoned]

bhosmer reviewed (commented):
I think moving implementation into its own space is great - clears up lots of confusion around filename similarity, snake case vs camel case, etc. 

I haven't made my way through the whole thing but for now have just hit one question, noted inline.

Also I'm curious, is it super easier to colocate the test code?

smessmer commented on aten/src/ATen/core/boxing/impl/unboxing_functor.h:
```diff
@@ -1,7 +1,7 @@
 #pragma once
```
I renamed it

smessmer committed:
Update on "Rename boxing/unboxing files and utilities"


The file and class names in ATen/core/boxing were quite confusing.
Let's rename them for readability.

Also move function schema inference out of the boxing logic into op_registration.h where it belongs.

This PR is just moving code around and renaming things. No logic changes.

Differential Revision: [D20653621](https://our.internmc.facebook.com/intern/diff/D20653621/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D20653621/)!

[ghstack-poisoned]

ezyang commented on aten/src/ATen/core/boxing/impl/wrap_function_into_runtime_functor.h:
```diff
@@ -0,0 +1,39 @@
+  // In the case of function pointers, since it is a runtime function pointer,
+  // there is an overhead for calling it whenever the kernel is invoked.
+  template<class FuncType>
+  using WrapFunctionIntoRuntimeFunctor = detail::WrapFunctionIntoRuntimeFunctor_<
```
Names... don't line up with filename?

smessmer replied:
ok I changed it to camel case. Makes it a bit weird because other files are snake case but we'll have to live with either the mismatch there or here.

smessmer committed:
Update on "Rename boxing/unboxing files and utilities"


The file and class names in ATen/core/boxing were quite confusing.
Let's rename them for readability.

Also move function schema inference out of the boxing logic into op_registration.h where it belongs.

This PR is just moving code around and renaming things. No logic changes.

Differential Revision: [D20653621](https://our.internmc.facebook.com/intern/diff/D20653621/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D20653621/)!

[ghstack-poisoned]

smessmer committed:
Update on "Rename boxing/unboxing files and utilities"


The file and class names in ATen/core/boxing were quite confusing.
Let's rename them for readability.

Also move function schema inference out of the boxing logic into op_registration.h where it belongs.

This PR is just moving code around and renaming things. No logic changes.

Differential Revision: [D20653621](https://our.internmc.facebook.com/intern/diff/D20653621/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D20653621/)!

[ghstack-poisoned]

smessmer committed:
Update on "Rename boxing/unboxing files and utilities"


The file and class names in ATen/core/boxing were quite confusing.
Let's rename them for readability.

Also move function schema inference out of the boxing logic into op_registration.h where it belongs.

This PR is just moving code around and renaming things. No logic changes.

Differential Revision: [D20653621](https://our.internmc.facebook.com/intern/diff/D20653621/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D20653621/)!

[ghstack-poisoned]

smessmer committed:
Update on "Rename boxing/unboxing files and utilities"


The file and class names in ATen/core/boxing were quite confusing.
Let's rename them for readability.

Also move function schema inference out of the boxing logic into op_registration.h where it belongs.

This PR is just moving code around and renaming things. No logic changes.

Differential Revision: [D20653621](https://our.internmc.facebook.com/intern/diff/D20653621/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D20653621/)!

[ghstack-poisoned]

smessmer committed:
Update on "Rename boxing/unboxing files and utilities"


The file and class names in ATen/core/boxing were quite confusing.
Let's rename them for readability.

Also move function schema inference out of the boxing logic into op_registration.h where it belongs.

This PR is just moving code around and renaming things. No logic changes.

Differential Revision: [D20653621](https://our.internmc.facebook.com/intern/diff/D20653621/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D20653621/)!

[ghstack-poisoned]

smessmer committed:
Update on "Rename boxing/unboxing files and utilities"


The file and class names in ATen/core/boxing were quite confusing.
Let's rename them for readability.

Also move function schema inference out of the boxing logic into op_registration.h where it belongs.

This PR is just moving code around and renaming things. No logic changes.

Differential Revision: [D20653621](https://our.internmc.facebook.com/intern/diff/D20653621/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D20653621/)!

[ghstack-poisoned]
